### PR TITLE
Populate and read from SyncStats table

### PR DESCRIPTION
**Pull request by alovew**
## What
This PR does a few things around the SyncStats table:

1. There is a new migration for updating the attempt_id foreign key from an Int to a BigInt, since the id on the attempts table is a BigInt
2. Updates the `writeOutput` method in DefaultJobPersistence.java. Because we are continuing to store sync stats in the JSON blob, I put the write to sync_stats inside the transaction that writes to the sync output to the attempts table. Once we move SyncStats completely out of that JSON blob, we should have a separate method for writing to the sync stats table.
3. New method getSyncStats for reading records from the SyncStats table

## Recommended reading order
1.  New migration: ChangeSyncStatsForeignKey
2. DefaultJobPersistence.java: updated write output method
4. DefaultJobPersistence.java: new getSyncStats method 
5. Tests

### CODE REVIEW

**Review by jdpgrailsdev — APPROVED** [review]
:shipit:

File: airbyte-scheduler/scheduler-persistence/src/main/java/io/airbyte/scheduler/persistence/DefaultJobPersistence.java [hunk 1]
```diff
@@ -305,14 +309,37 @@ public Optional<String> getAttemptTemporalWorkflowId(final long jobId, final int
...
+      final Long attemptId = record.get().get("id", Long.class);
+
+      ctx.insertInto(SYNC_STATS)
+          .set(SYNC_STATS.ID, UUID.randomUUID())
+          .set(SYNC_STATS.UPDATED_AT, now)
+          .set(SYNC_STATS.CREATED_AT, now)
+          .set(SYNC_STATS.ATTEMPT_ID, attemptId)
+          .set(SYNC_STATS.BYTES_EMITTED, syncStats.getBytesEmitted())
+          .set(SYNC_STATS.RECORDS_EMITTED, syncStats.getRecordsEmitted())
+          .set(SYNC_STATS.RECORDS_COMMITTED, syncStats.getRecordsEmitted())
```
Comment (gosusnp) [inline airbyte-scheduler/scheduler-persistence/src/main/java/io/airbyte/scheduler/persistence/DefaultJobPersistence.java:332]: `getRecordsCommitted`?

File: airbyte-scheduler/scheduler-persistence/src/test/java/io/airbyte/scheduler/persistence/DefaultJobPersistenceTest.java [hunk 2]
```diff
@@ -249,14 +252,34 @@ void testWriteOutput() throws IOException {
     final long jobId = jobPersistence.enqueueJob(SCOPE, SPEC_JOB_CONFIG).orElseThrow();
     final int attemptNumber = jobPersistence.createAttempt(jobId, LOG_PATH);
     final Job created = jobPersistence.getJob(jobId);
-    final JobOutput jobOutput = new JobOutput().withOutputType(JobOutput.OutputType.DISCOVER_CATALOG);
+    final SyncStats syncStats =
+        new SyncStats().withBytesEmitted(100L).withRecordsEmitted(10L).withRecordsCommitted(10L).withDestinationStateMessagesEmitted(1L)
```
Comment (gosusnp) [inline airbyte-scheduler/scheduler-persistence/src/test/java/io/airbyte/scheduler/persistence/DefaultJobPersistenceTest.java:256]: Different values for `recordsEmitted` and `recordsCommitted` would have caught the typo.

**Review by gosusnp — DISMISSED** [review]
Looks good overall.

**Pull request merged**